nwb_conversion_tools/ecephys/ecephys.py

```diff
@@ -70,22 +70,86 @@ def add_units(self):
-                                                         feature_name='waveforms')
-                    relevant_ch = most_relevant_ch(wf)
-                    # Spike traces on the most relevant channel
-                    traces = wf[:, relevant_ch, :]
-                    traces_avg = np.mean(traces, axis=0)
-                    traces_std = np.std(traces, axis=0)
-                    self.nwbfile.add_unit(
-                        id=id,
-                        spike_times=spkt,
-                        waveform_mean=traces_avg,
-                        waveform_sd=traces_std
-                    )
+                    if 'electrode_group' in self.SX.get_unit_property_names(unit_id=id):
+                        # Stores average and std of spike traces
+                        wf = self.SX.get_unit_spike_features(unit_id=id,
+                                                             feature_name='waveforms')
+                        relevant_ch = most_relevant_ch(wf)
+                        # Spike traces on the most relevant channel
+                        traces = wf[:, relevant_ch, :]
+                        traces_avg = np.mean(traces, axis=0)
+                        traces_std = np.std(traces, axis=0)
+                        self.nwbfile.add_unit(
+                            id=id,
+                            spike_times=spkt,
+                            waveform_mean=traces_avg,
+                            waveform_sd=traces_std,
+                            electrode_group=self.SX.get_unit_property(id,'electrode_group')
+                        )
+                    else:
+                        # Stores average and std of spike traces
+                        wf = self.SX.get_unit_spike_features(unit_id=id,
+                                                             feature_name='waveforms')
+                        relevant_ch = most_relevant_ch(wf)
+                        # Spike traces on the most relevant channel
+                        traces = wf[:, relevant_ch, :]
+                        traces_avg = np.mean(traces, axis=0)
+                        traces_std = np.std(traces, axis=0)
+                        self.nwbfile.add_unit(
+                            id=id,
+                            spike_times=spkt,
+                            waveform_mean=traces_avg,
+                            waveform_sd=traces_std
+                        )
                 else:
-                    self.nwbfile.add_unit(id=id, spike_times=spkt)
+                    if 'electrode_group' in self.SX.get_unit_property_names(unit_id=id):
+                        self.nwbfile.add_unit(id=id, spike_times=spkt, electrode_group=self.SX.get_unit_property(id,'electrode_group'))
+                    else: 
+                        self.nwbfile.add_unit(id=id, spike_times=spkt)
+        # Extract and add custom column data from unit properties
+        descriptions = [
+            {
```

Comment:
This class should be generic to all SpikeExtractors, not specific to Neuroscope, so I'd rather not have this

Reply:
Fixed! Now assumes the unit property is a dictionary with all the relevant information specific to that dataset and unit.